opus/application/static_media/js/search.js

```diff
@@ -1204,40 +1204,59 @@ var o_search = {
-                            $(id).html(`<span class="${hintsTextClass}">` + mults[value] + "</span>");
+                            $(id).html(`<span class="${hintsTextClass}" data-value=${mults[value]}>` + mults[value] + "</span>");
                             if ($(id).parent().hasClass("fadey")) {
                                 $(id).parent().removeClass("fadey");
                             }
                         } else {
-                            $(id).html(`<span class="${hintsTextClass}">0</span>`);
+                            $(id).html(`<span class="${hintsTextClass}" data-value=0>0</span>`);
                             $(id).parent().addClass("fadey");
                         }
                     } else {
                         // Display "--" next to surfacegeometrytargetname inputs if the input is not selected.
                         $(id).parent().removeClass("fadey");
                         if (mults[value]) {
                             if ($(this).is(":checked")) {
-                                $(id).html(`<span class="${hintsTextClass}">` + mults[value] + "</span>");
+                                $(id).html(`<span class="${hintsTextClass}" data-value=${mults[value]}>` + mults[value] + "</span>");
                             } else {
-                                $(id).html(`<span class="${hintsTextClass}">--</span>`);
+                                $(id).html(`<span class="${hintsTextClass}" data-value="--">--</span>`);
                             }
                         } else {
                             if ($(this).is(":checked")) {
-                                $(id).html(`<span class="${hintsTextClass}">0</span>`);
-
+                                $(id).html(`<span class="${hintsTextClass}" data-value=0>0</span>`);
                             } else {
-                                $(id).html(`<span class="${hintsTextClass}">--</span>`);
+                                $(id).html(`<span class="${hintsTextClass}" data-value="--">--</span>`);
                             }
                         }
                     }
                 });
+
+                // Count the total hints under each group name and display them in the ui
+                // If the hints are all "--" under a group, we will display "--" next to
+                // the group name.
+                $("#" + widget + " .mult_group").each( function() {
+                    let sum = 0;
+                    // The flag used to determine if we will display the total hints or "--"
+                    let displaySum = false;
+                    let group = $(this).data("group");
+                    let groupClass = `.mult_group_${group}`;
+
+                    $("#" + widget + ` .mult_group[data-group="${group}"] .${hintsTextClass}`).each(function() {
+                        let multVal = $(this).data("value");
+                        if (!displaySum && !isNaN(parseInt(multVal))) { displaySum = true; }
```

Comment:
Perhaps only evaluate parseInt(multVal) once into a variable? Also no real reason to check for !displaySum first. You can just always set it to true. Since your second if is checking for NaN, it seems like this could be simpler as:

```
let multValInt = parseInt(multVal);
if (!isNaN(multValInt) {
    displaySum = true;
    sum += multValInt;
}
```